### DISCUSSION TITLE
Welcome to SimpleWaterfall-PowerBI Discussions!

### DISCUSSION THREAD

**Discussion by nishantjainuk** [nishantjainuk/SimpleWaterfall-PowerBI#48] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by lynnzu9** [comment]
Hi Nishant, 

Hope this message finds you well and safe during this pandemic. 

Just dropping by to say thank your so much for your simple waterfall chart in Power BI, it's exactly what I wanted! Love the easily created bridge between two tables. 

I see it works great with positive numbers, while if the numbers are negative, the categories/bridges sits at the bottom. Of course that's because negative are below zero line. Not sure if there could be a function to shift the Y/zero axis so all waterfall will show in same direction(on top) no matter negative or positive.   

But anyway, huge thank you again! 

Take care, 
Lin

**Reply by nishantjainuk** [reply]
@lynnzu9 , thanks for your suggestion. Would it be possible to send me a screenshot of an example of how you think it might work. I am struggling to visualise it

**Reply by lynnzu9** [reply]
HI Nishant,

Thanks for you reply, I attached a example here. 

In this screenshot, all numbers are negative, e.g. expenses. and the category bridge sits in bottom. Whereas if the numbers are positive, the bridges sit on top, which is perfect.  So i'm wondering if the zero axis can be customized like in Power Point Chart, so we can have all the bridges on top even with negative numbers. Not sure if that makes sense? 

Best Regards,
Lin
<img width="537" alt="Capture" src="https://user-images.githubusercontent.com/93356187/139610160-f707d2bc-7f37-4020-b8cd-f86d71f17271.PNG">

**Reply by nishantjainuk** [reply]
@lynnzu9 

It won't be possible to do this as an option because mathematically it won't work. I do think that you should be able to do this using DAX.

Below is the same example that you had using the simple waterfall

![image](https://user-images.githubusercontent.com/24504997/142744580-cac4377b-798a-4a68-8e8d-5cf49dafc617.png)

Below are the steps I have used to create this:

1. Change the sign by multiplying the value by -1
2. Step one will convert all negative to positive. To show them as negative, I have used custom format "-0;0". You can use your 
![image](https://user-images.githubusercontent.com/24504997/142744584-89839fe7-6926-4599-b9c9-c624168d51a5.png)

I hope this works for you. Let me know how it goes

Thanks
Nishant

**Comment by lynnzu9** [comment]
Hi Nishant, 

Thanks so much for your reply. Your method is a perfect solution! 

I created 2 more measures for this waterfall chart to give me a positive total expense budget and actual. Then applied the measures with custom format -#,##0;#,##0. 

Thanks, life saver!

Best regards,
Lin

**Comment by blevib** [comment]
Hi Nishant, 

I recently found your Simple Waterfall Power BI chart. First of all a massive thank you, it is a great tool! 

There is one thing I cannot work out how to set up, or if it is available at all and I wonder if you might be able to help with it.

I have two measures in my report:
Order Qty (a dynamic calculation)
Delivered Qty (another dynamic calculation)

I have product categories: PackTechSR.

I would like to see the top 10 PackTechSR categories which are ranked based on the difference between Order Qty and Delivered Qty, and then the rest grouped as Others.
At the moment I only see option to sort them either by Order Qty or by Delivery quantity, but not by the difference between the 2 (which are the waterfall step values).

![PBI waterfall-1](https://user-images.githubusercontent.com/106593923/171162210-4e875c82-9af4-437b-99a1-5a4604c31fe7.png)
![PBI waterfall-2](https://user-images.githubusercontent.com/106593923/171162214-0071f725-7ae0-41a7-8ede-71eee4538c57.png)
![PBI waterfall-3](https://user-images.githubusercontent.com/106593923/171162215-68150450-4475-40e5-9dea-d7da4af12a38.png)

If you could give me some directions, or consider it for a later release as new feature that would be so much appreciated.

Many thanks

**Comment by bruno-idf** [comment]
Hi Nishant,

Thanks for developing this fantastic tool!

The chart works perfectly on the Power BI desktop, but when I publish it, it breaks when I try to see it in focus mode. I reinstalled it, but the error persists. Do you know how can I fix it? See the images below
![image](https://user-images.githubusercontent.com/122756653/212572558-e2909adb-23d8-4f0b-afb7-b5009fd6e8bc.png)

When I try to open the focus mode it breaks:
![image](https://user-images.githubusercontent.com/122756653/212572606-52aaef58-b26c-4367-a462-23b4870fc41d.png)

I have to click on the double arrow icon to drill it down and then it works.

**Comment by railfan727** [comment]
How would I create a graph with 3 totals bars like the one shown here?

![image](https://user-images.githubusercontent.com/124288797/216414770-17b28269-8a48-4b6a-8f50-79bca81ad455.png)

If I put three measures in the Values field, it attempts to break the differences between pillar 2 and pillar 3 along the same category definitions as for pillar 1 to pillar 2... but this is not logically appropriate for what I'm trying to do. How can I get three pillars but have the variances between each of them treated separately?

**Comment by aakritigupta01** [comment]
Hi!

I am trying to use the simple waterfall for one of my dashboards and it works great on the desktop, but on the service, every time I refresh the page, the charts disappear. When I click on edit and switch back and forth, it appears again, but how can I make sure that everyone can else can simply go to the page and see the visuals? Is there a setting that I might be missing?

**Comment by JessH01** [comment]
Hi there, this waterfall is GREAT!  One thing we are stuck on - my data set is formatted as currency, but that formatting disappears when I use your visual.  Am I doing something wrong?  I have a mix of costs and counts on the same dash, so it matters that people can clearly see which is which.  Thanks!

**Comment by Pedroheferreira** [comment]
Hello, would it be possible for me to put the total value on the left and the graphs be on the right in decreasing order?

**Comment by nishantjainuk** [comment]
Yes it should be possible. There is an option to define sort order in the visual properties.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by Pedroheferreira** [reply]
Could  I can put the total in left  in the first? 
![image](https://github.com/nishantjainuk/SimpleWaterfall-PowerBI/assets/57763197/8257672a-db0d-447a-94aa-fcbc50b88774)

**Comment by nishantjainuk** [comment]
I think it should be. There should be an option to sort the data

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by Pedroheferreira** [reply]
It wasn't possible, I thought it would be possible if we could change something in the code

**Reply by Pedroheferreira** [reply]
Cumulative total in Left not Right

**Comment by nishantjainuk** [comment]
Agreed, can't do total on the left for cumulative total. Let me add it to my list of things to do in my next release

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by Pedroheferreira** [reply]
Thank you very much for your attention and commitment. I wish I could help or learn from you in some way, if possible I would like to take some classes with you :) Congratulations on the work

**Comment by samdewar** [comment]
Hey, Loving this chart

I was wondering if anyone knows how to exclude values below a certain threshold
e.g.
![image](https://github.com/user-attachments/assets/5c9c761d-e73e-40a4-8d63-289a8969992a)

With filters applied:
![image](https://github.com/user-attachments/assets/2bee811c-2597-4021-8133-9567bc08699e)

I'd like the pillars with height=0 to disappear
Is this possible?

**Comment by Billytoh** [comment]
Hello,

The simple waterfall chart's filter function has been disabled after the new updates. Can you take a look? Thank you in advance!

**Comment by Saydullayev-Oybek** [comment]
https://github.com/user-attachments/assets/4e6acb3d-8c77-46a4-b4d7-dc85ab9069f9

When i choose horizontal option the the chart is become non-interactive i mean when i click on bar it is not highlighting but it is working with vertical option. I have no idea why is this happening for horizontal option
